refactor(application model refinement)

ChrOertlin opened the pull request:
## Description

### Added

- Tests for Application model functions

### Changed

- Function naming
- Refactored statusDB access to go via filter functions
- Refactored applications() into multiple functions
- Refactored applications() calls into new function calls of smaller functions

### Fixed

- MockApplication to be iterable to conform to new List[Application] return type of get_applications() function


### How to prepare for test

- [x] Ssh to relevant server (depending on type of change)
- [x] Use stage: `us`
- [x] Paxa the environment: `paxa`
- [x] Install on stage (example for Hasta):
    ```shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_main -t cg -b refactor_refine_application -a
    ```

### How to test

- [x] Do `cg import application`
- [x] Do `cg add`
- [x] `cg upload vogue apptags`

- [x] Check that the order form can fetch applications

### Expected test outcome

- [x] Check that ...
- [x] Take a screenshot and attach or copy/paste the output.

## Review

- [ ] Tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [x] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan

- [ ] Document in ...
- [ ] Deploy this branch on ...
- [ ] Inform to ...

ChrOertlin commented:
Can select application in orderform:

![image](https://user-images.githubusercontent.com/8860786/225009438-0b64c283-7352-4d86-aec9-c0fa3fde1632.png)

henrikstranneheim commented on tests/store/api/test_findbasicdata.py:
```diff
@@ -123,6 +121,78 @@ def test_get_application_query(base_store: Store):
+    assert (application.is_archived is False for application in applications)
+
+
+def test_get_applications_by_prep_category(microbial_store: Store, prep_category="mic"):
```
Use constant

diitaz93 replied:
Or fixture?

henrikstranneheim commented on tests/store/api/test_findbasicdata.py:
```diff
@@ -123,6 +121,78 @@ def test_get_application_query(base_store: Store):
+    applications: List[Application] = microbial_store.get_applications_is_not_archived()
+
+    # THEN return a application with the supplied application tag
+    assert len(applications) == 4
```
```suggestion
    assert len(applications) == EXPECTED_NR_ARCHIVED_APPLICATIONS
```

henrikstranneheim commented on tests/store/api/test_findbasicdata.py:
```diff
@@ -123,6 +121,78 @@ def test_get_application_query(base_store: Store):
+    )
+
+    # THEN return a application with the supplied application tag
+    assert len(applications) == 7
```
```suggestion
    assert len(applications) == EXPECTED_NR_OF_PREP_CATEGORY_APPS
```

henrikstranneheim commented on tests/store/api/test_findbasicdata.py:
```diff
@@ -123,6 +121,78 @@ def test_get_application_query(base_store: Store):
+def test_get_applications_by_prep_category_and_is_not_archived(
+    microbial_store: Store, prep_category="mic"
+):
+    """Test function to return the application by prep category."""
```
```suggestion
    """Test function to return the application by prep category and not archived."""
```

henrikstranneheim commented on tests/store/api/test_findbasicdata.py:
```diff
@@ -123,6 +121,78 @@ def test_get_application_query(base_store: Store):
+
+
+def test_get_applications(microbial_store: Store):
+    """Test function to return the application by prep category."""
```
```suggestion
    """Test function to return the applications."""
```

henrikstranneheim commented on tests/store/api/test_findbasicdata.py:
```diff
@@ -123,6 +121,78 @@ def test_get_application_query(base_store: Store):
+
+
+def test_get_applications_is_not_archived(microbial_store: Store):
+    """Test function to return the application by tag."""
```
```suggestion
    """Test function to return the application when not archived."""
```

ChrOertlin commented:
`cg import application test_application.xlsx CO`
/home/proj/stage/bin/miniconda3/envs/S_main/lib/python3.7/site-packages/sqlalchemy/orm/relationships.py:2231: SAWarning: Cascade settings "delete, merge, save-update" should not be combined with a viewonly=True relationship.   This configuration will raise an error in version 1.4.  Note that in versions prior to 1.4, these cascade settings may still produce a mutating effect even though this relationship is marked as viewonly=True.
  "viewonly=True." % (", ".join(sorted(non_viewonly)))
/home/proj/stage/bin/miniconda3/envs/S_main/lib/python3.7/site-packages/sqlalchemy/orm/relationships.py:1998: SAWarning: Setting backref / back_populates on relationship Demux.samples to refer to viewonly relationship Sample.demuxes should include sync_backref=False set on the Demux.samples relationship.  (this warning may be suppressed after 10 occurrences)
  (rel_b, rel_a, rel_b),
/home/proj/stage/bin/miniconda3/envs/S_main/lib/python3.7/site-packages/sqlalchemy/orm/relationships.py:1998: SAWarning: Setting backref / back_populates on relationship Sample.demuxes to refer to viewonly relationship Demux.samples should include sync_backref=False set on the Sample.demuxes relationship.  (this warning may be suppressed after 10 occurrences)
  (rel_b, rel_a, rel_b),
adding new application to transaction A
all applications successfully added to transaction, committing transaction

henrikstranneheim reviewed (commented):
Awesome - did not have time to finish so will only comment

ChrOertlin commented:
`cg add sample -s MALE -a WGSPCFC030 cust000 testytest`
/home/proj/stage/bin/miniconda3/envs/S_main/lib/python3.7/site-packages/sqlalchemy/orm/relationships.py:2231: SAWarning: Cascade settings "delete, merge, save-update" should not be combined with a viewonly=True relationship.   This configuration will raise an error in version 1.4.  Note that in versions prior to 1.4, these cascade settings may still produce a mutating effect even though this relationship is marked as viewonly=True.
  "viewonly=True." % (", ".join(sorted(non_viewonly)))
/home/proj/stage/bin/miniconda3/envs/S_main/lib/python3.7/site-packages/sqlalchemy/orm/relationships.py:1998: SAWarning: Setting backref / back_populates on relationship Demux.samples to refer to viewonly relationship Sample.demuxes should include sync_backref=False set on the Demux.samples relationship.  (this warning may be suppressed after 10 occurrences)
  (rel_b, rel_a, rel_b),
/home/proj/stage/bin/miniconda3/envs/S_main/lib/python3.7/site-packages/sqlalchemy/orm/relationships.py:1998: SAWarning: Setting backref / back_populates on relationship Sample.demuxes to refer to viewonly relationship Demux.samples should include sync_backref=False set on the Sample.demuxes relationship.  (this warning may be suppressed after 10 occurrences)
  (rel_b, rel_a, rel_b),
merelynovelcub: new sample added

![image](https://user-images.githubusercontent.com/8860786/225029150-1ac7fd9c-1ce8-48d5-8291-07addfe09be6.png)

ChrOertlin commented:
cg upload vogue apptags
/home/proj/stage/bin/miniconda3/envs/S_main/lib/python3.7/site-packages/sqlalchemy/orm/relationships.py:2231: SAWarning: Cascade settings "delete, merge, save-update" should not be combined with a viewonly=True relationship.   This configuration will raise an error in version 1.4.  Note that in versions prior to 1.4, these cascade settings may still produce a mutating effect even though this relationship is marked as viewonly=True.
  "viewonly=True." % (", ".join(sorted(non_viewonly)))
Called undefined __fields__ on HousekeeperAPI, please wrap
Called undefined __fields__ on HousekeeperAPI, please wrap
Called undefined __fields__ on HousekeeperAPI, please wrap
Called undefined __fields__ on HousekeeperAPI, please wrap
Called undefined __fields__ on HousekeeperAPI, please wrap
Called undefined __fields__ on HousekeeperAPI, please wrap
Called undefined __fields__ on HousekeeperAPI, please wrap
Called undefined __fields__ on HousekeeperAPI, please wrap
Called undefined __fields__ on HousekeeperAPI, please wrap
Called undefined __fields__ on HousekeeperAPI, please wrap
----------------- UPLOAD -----------------
Called undefined __fields__ on HousekeeperAPI, please wrap
Called undefined __fields__ on HousekeeperAPI, please wrap
----------------- TRENDING -----------------------
----------------- APPLICATION TAGS -----------------------
/home/proj/stage/bin/miniconda3/envs/S_main/lib/python3.7/site-packages/sqlalchemy/orm/relationships.py:1998: SAWarning: Setting backref / back_populates on relationship Demux.samples to refer to viewonly relationship Sample.demuxes should include sync_backref=False set on the Demux.samples relationship.  (this warning may be suppressed after 10 occurrences)
  (rel_b, rel_a, rel_b),
/home/proj/stage/bin/miniconda3/envs/S_main/lib/python3.7/site-packages/sqlalchemy/orm/relationships.py:1998: SAWarning: Setting backref / back_populates on relationship Sample.demuxes to refer to viewonly relationship Demux.samples should include sync_backref=False set on the Sample.demuxes relationship.  (this warning may be suppressed after 10 occurrences)
  (rel_b, rel_a, rel_b),
Running command /home/proj/stage/bin/miniconda3/envs/S_vogue/bin/vogue --config /home/proj/stage/servers/config/hasta.scilifelab.se/vogue-stage.yaml load apptag [{"tag": "RMLP10R825", "prep_category": ""}, {"tag": "VWGDPTR001", "prep_category": "cov"}, {"tag": "VWXONTR000", "prep_category": "cov"}, {"tag": "MWGNXBR002", "prep_category": "mic"}, {"tag": "MWGNXBR003", "prep_category": "mic"}, {"tag": "MWGNXMR002", "prep_category": "mic"}, {"tag": "MWGNXMR003", "prep_category": "mic"}, {"tag": "MWGNXTR003", "prep_category": "mic"}, {"tag": "MWGPCFR002", "prep_category": "mic"}, {"tag": "MWLNXBR002", "prep_category": "mic"}, {"tag": "MWLNXBR003", "prep_category": "mic"}, {"tag": "MWLNXMR002", "prep_category": "mic"}, {"tag": "MWLNXMR003", "prep_category": "mic"}, {"tag": "MWLNXTR002", "prep_category": "mic"}, {"tag": "MWLNXTR003", "prep_category": "mic"}, {"tag": "MWLPCFR002", "prep_category": "mic"}, {"tag": "MWMNXBR002", "prep_category": "mic"}, {"tag": "MWMNXBR003", "prep_category": "mic"}, {"tag": "MWMNXMR002", "prep_category": "mic"}, {"tag": "MWMNXMR003", "prep_category": "mic"}, {"tag": "MWMNXTR002", "prep_category": "mic"}, {"tag": "MWMNXTR003", "prep_category": "mic"}, {"tag": "MWRNXTR003", "prep_category": "mic"}, {"tag": "MWXNXBR002", "prep_category": "mic"}, {"tag": "MWXNXBR003", "prep_category": "mic"}, {"tag": "MWXNXMR002", "prep_category": "mic"}, {"tag": "MWXNXMR003", "prep_category": "mic"}, {"tag": "MWXNXTR002", "prep_category": "mic"}, {"tag": "MWXNXTR003", "prep_category": "mic"}, {"tag": "MWXPCFR000", "prep_category": "mic"}, {"tag": "VWGNXTK500", "prep_category": "mic"}, {"tag": "VWGNXTR001", "prep_category": "mic"}, {"tag": "VWLNXTR001", "prep_category": "mic"}, {"tag": "VWMNXTK500", "prep_category": "mic"}, {"tag": "DEVP10R005", "prep_category": "rml"}, {"tag": "DEVP10R008", "prep_category": "rml"}, {"tag": "DEVP10R010", "prep_category": "rml"}, {"tag": "DEVP10R030", "prep_category": "rml"}, {"tag": "NIPCUSR300", "prep_category": "rml"}, {"tag": "RMLCUSR800", "prep_category": "rml"}, {"tag": "RMLCUSS160", "prep_category": "rml"}, {"tag": "RMLP05R800", "prep_category": "rml"}, {"tag": "RMLP07R025", "prep_category": "rml"}, {"tag": "RMLP10R010", "prep_category": "rml"}, {"tag": "RMLP10R020", "prep_category": "rml"}, {"tag": "RMLP10R040", "prep_category": "rml"}, {"tag": "RMLP10R150", "prep_category": "rml"}, {"tag": "RMLP10R300", "prep_category": "rml"}, {"tag": "RMLP10S130", "prep_category": "rml"}, {"tag": "RMLP15C030", "prep_category": "rml"}, {"tag": "RMLP15R015", "prep_category": "rml"}, {"tag": "RMLP15R020", "prep_category": "rml"}, {"tag": "RMLP15R100", "prep_category": "rml"}, {"tag": "RMLP15R200", "prep_category": "rml"}, {"tag": "RMLP15R300", "prep_category": "rml"}, {"tag": "RMLP15R325", "prep_category": "rml"}, {"tag": "RMLP15R400", "prep_category": "rml"}, {"tag": "RMLP15R500", "prep_category": "rml"}, {"tag": "RMLP15R750", "prep_category": "rml"}, {"tag": "RMLP15R825", "prep_category": "rml"}, {"tag": "RMLP15S100", "prep_category": "rml"}, {"tag": "RMLP15S125", "prep_category": "rml"}, {"tag": "RMLP15S150", "prep_category": "rml"}, {"tag": "RMLP15S175", "prep_category": "rml"}, {"tag": "RMLP15S200", "prep_category": "rml"}, {"tag": "RMLP15S225", "prep_category": "rml"}, {"tag": "RMLP15S250", "prep_category": "rml"}, {"tag": "RMLP15S275", "prep_category": "rml"}, {"tag": "RMLP15S300", "prep_category": "rml"}, {"tag": "RMLP15S325", "prep_category": "rml"}, {"tag": "RMLP15S350", "prep_category": "rml"}, {"tag": "RMLP15S375", "prep_category": "rml"}, {"tag": "RMLP15S400", "prep_category": "rml"}, {"tag": "RMLP15S425", "prep_category": "rml"}, {"tag": "RMLP15S450", "prep_category": "rml"}, {"tag": "RMLP15S475", "prep_category": "rml"}, {"tag": "RMLP15S500", "prep_category": "rml"}, {"tag": "RMLP25R015", "prep_category": "rml"}, {"tag": "RMLP30R025", "prep_category": "rml"}, {"tag": "RMLS05R150", "prep_category": "rml"}, {"tag": "RMLS05R200", "prep_category": "rml"}, {"tag": "RMLS05R300", "prep_category": "rml"}, {"tag": "CCPSLIR005", "prep_category": "tgs"}, {"tag": "EFTSLIR020", "prep_category": "tgs"}, {"tag": "EFTSLIR030", "prep_category": "tgs"}, {"tag": "EFTSXTR020", "prep_category": "tgs"}, {"tag": "EFTSXTR030", "prep_category": "tgs"}, {"tag": "EXLTTTR025", "prep_category": "tgs"}, {"tag": "EXOTTTR050", "prep_category": "tgs"}, {"tag": "MHPSLIR025", "prep_category": "tgs"}, {"tag": "MHPSXTR025", "prep_category": "tgs"}, {"tag": "PALKTTR010", "prep_category": "tgs"}, {"tag": "PALKTTR020", "prep_category": "tgs"}, {"tag": "PALKTTR030", "prep_category": "tgs"}, {"tag": "PALKTTR040", "prep_category": "tgs"}, {"tag": "PALKTTR050", "prep_category": "tgs"}, {"tag": "PALKTTR060", "prep_category": "tgs"}, {"tag": "PALKTTR100", "prep_category": "tgs"}, {"tag": "PALTTTR010", "prep_category": "tgs"}, {"tag": "PALTTTR020", "prep_category": "tgs"}, {"tag": "PALTTTR030", "prep_category": "tgs"}, {"tag": "PALTTTR040", "prep_category": "tgs"}, {"tag": "PALTTTR050", "prep_category": "tgs"}, {"tag": "PANKTTR010", "prep_category": "tgs"}, {"tag": "PANKTTR015", "prep_category": "tgs"}, {"tag": "PANKTTR020", "prep_category": "tgs"}, {"tag": "PANKTTR030", "prep_category": "tgs"}, {"tag": "PANKTTR040", "prep_category": "tgs"}, {"tag": "PANKTTR050", "prep_category": "tgs"}, {"tag": "PANKTTR060", "prep_category": "tgs"}, {"tag": "PANKTTR080", "prep_category": "tgs"}, {"tag": "PANKTTR100", "prep_category": "tgs"}, {"tag": "PANKTTR200", "prep_category": "tgs"}, {"tag": "EXLKTTR020", "prep_category": "wes"}, {"tag": "EXLKTTR040", "prep_category": "wes"}, {"tag": "EXLKTTR060", "prep_category": "wes"}, {"tag": "EXLKTTR080", "prep_category": "wes"}, {"tag": "EXLTTTR020", "prep_category": "wes"}, {"tag": "EXLTTTR040", "prep_category": "wes"}, {"tag": "EXLTTTR060", "prep_category": "wes"}, {"tag": "EXLTTTR080", "prep_category": "wes"}, {"tag": "EXOKTTR020", "prep_category": "wes"}, {"tag": "EXOKTTR030", "prep_category": "wes"}, {"tag": "EXOKTTR040", "prep_category": "wes"}, {"tag": "EXOKTTR060", "prep_category": "wes"}, {"tag": "EXOKTTR080", "prep_category": "wes"}, {"tag": "EXOKTTR100", "prep_category": "wes"}, {"tag": "EXOKTTR120", "prep_category": "wes"}, {"tag": "EXOKTTR140", "prep_category": "wes"}, {"tag": "EXOKTTR200", "prep_category": "wes"}, {"tag": "EXOKTTR500", "prep_category": "wes"}, {"tag": "EXOQXTR030", "prep_category": "wes"}, {"tag": "EXOQXTR075", "prep_category": "wes"}, {"tag": "EXOQXTR100", "prep_category": "wes"}, {"tag": "EXORMLR030", "prep_category": "wes"}, {"tag": "EXOSFFR030", "prep_category": "wes"}, {"tag": "EXOSFFR050", "prep_category": "wes"}, {"tag": "EXOSFFR060", "prep_category": "wes"}, {"tag": "EXOSFFR075", "prep_category": "wes"}, {"tag": "EXOSFFR100", "prep_category": "wes"}, {"tag": "EXOSFFR150", "prep_category": "wes"}, {"tag": "EXOSLIR020", "prep_category": "wes"}, {"tag": "EXOSLIR025", "prep_category": "wes"}, {"tag": "EXOSLIR030", "prep_category": "wes"}, {"tag": "EXOSLIR040", "prep_category": "wes"}, {"tag": "EXOSLIR050", "prep_category": "wes"}, {"tag": "EXOSLIR060", "prep_category": "wes"}, {"tag": "EXOSLIR075", "prep_category": "wes"}, {"tag": "EXOSLIR100", "prep_category": "wes"}, {"tag": "EXOSLIR150", "prep_category": "wes"}, {"tag": "EXOSXTR020", "prep_category": "wes"}, {"tag": "EXOSXTR025", "prep_category": "wes"}, {"tag": "EXOSXTR030", "prep_category": "wes"}, {"tag": "EXOSXTR040", "prep_category": "wes"}, {"tag": "EXOSXTR045", "prep_category": "wes"}, {"tag": "EXOSXTR050", "prep_category": "wes"}, {"tag": "EXOSXTR060", "prep_category": "wes"}, {"tag": "EXOSXTR075", "prep_category": "wes"}, {"tag": "EXOSXTR100", "prep_category": "wes"}, {"tag": "EXOSXTR150", "prep_category": "wes"}, {"tag": "EXOTTTR020", "prep_category": "wes"}, {"tag": "EXOTTTR025", "prep_category": "wes"}, {"tag": "EXOTTTR030", "prep_category": "wes"}, {"tag": "EXOTTTR040", "prep_category": "wes"}, {"tag": "EXOTTTR060", "prep_category": "wes"}, {"tag": "EXOTTTR080", "prep_category": "wes"}, {"tag": "EXOTTTR100", "prep_category": "wes"}, {"tag": "EXOTTTR120", "prep_category": "wes"}, {"tag": "EXOTTTR125", "prep_category": "wes"}, {"tag": "EXOTTTR140", "prep_category": "wes"}, {"tag": "EXTKTTR020", "prep_category": "wes"}, {"tag": "EXTKTTR040", "prep_category": "wes"}, {"tag": "EXTSFFR030", "prep_category": "wes"}, {"tag": "EXTSFFR060", "prep_category": "wes"}, {"tag": "EXTSFFR100", "prep_category": "wes"}, {"tag": "EXTSLIR020", "prep_category": "wes"}, {"tag": "EXTSLIR030", "prep_category": "wes"}, {"tag": "EXTSLIR045", "prep_category": "wes"}, {"tag": "EXTSLIR060", "prep_category": "wes"}, {"tag": "EXTSLIR075", "prep_category": "wes"}, {"tag": "EXTSLIR100", "prep_category": "wes"}, {"tag": "EXTSXTR025", "prep_category": "wes"}, {"tag": "EXTSXTR030", "prep_category": "wes"}, {"tag": "EXTSXTR045", "prep_category": "wes"}, {"tag": "EXTSXTR060", "prep_category": "wes"}, {"tag": "EXTSXTR075", "prep_category": "wes"}, {"tag": "EXTSXTR100", "prep_category": "wes"}, {"tag": "EXTTTTR020", "prep_category": "wes"}, {"tag": "EXTTTTR025", "prep_category": "wes"}, {"tag": "EXTTTTR030", "prep_category": "wes"}, {"tag": "EXTTTTR040", "prep_category": "wes"}, {"tag": "EXTTTTR050", "prep_category": "wes"}, {"tag": "EXXCUSR000", "prep_category": "wes"}, {"tag": "PANTTTR010", "prep_category": "wes"}, {"tag": "PANTTTR015", "prep_category": "wes"}, {"tag": "PANTTTR020", "prep_category": "wes"}, {"tag": "PANTTTR030", "prep_category": "wes"}, {"tag": "PANTTTR040", "prep_category": "wes"}, {"tag": "PANTTTR050", "prep_category": "wes"}, {"tag": "A", "prep_category": "wgs"}, {"tag": "EXTSXTR020", "prep_category": "wgs"}, {"tag": "MELLIFR010", "prep_category": "wgs"}, {"tag": "MELLIFR020", "prep_category": "wgs"}, {"tag": "MELLIFR030", "prep_category": "wgs"}, {"tag": "MELNXTR010", "prep_category": "wgs"}, {"tag": "MELNXTR020", "prep_category": "wgs"}, {"tag": "MELNXTR030", "prep_category": "wgs"}, {"tag": "MELPCFR010", "prep_category": "wgs"}, {"tag": "MELPCFR020", "prep_category": "wgs"}, {"tag": "MELPCFR030", "prep_category": "wgs"}, {"tag": "METDPFR030", "prep_category": "wgs"}, {"tag": "METDPTR030", "prep_category": "wgs"}, {"tag": "METKAPR050", "prep_category": "wgs"}, {"tag": "METLIFR010", "prep_category": "wgs"}, {"tag": "METLIFR020", "prep_category": "wgs"}, {"tag": "METLIFR030", "prep_category": "wgs"}, {"tag": "METNXTR010", "prep_category": "wgs"}, {"tag": "METNXTR020", "prep_category": "wgs"}, {"tag": "METNXTR030", "prep_category": "wgs"}, {"tag": "METPCFR010", "prep_category": "wgs"}, {"tag": "METPCFR020", "prep_category": "wgs"}, {"tag": "METPCFR030", "prep_category": "wgs"}, {"tag": "METPCFR050", "prep_category": "wgs"}, {"tag": "WGLLIFC030", "prep_category": "wgs"}, {"tag": "WGLLIFS120", "prep_category": "wgs"}, {"tag": "WGLPCFC030", "prep_category": "wgs"}, {"tag": "WGLPCFS120", "prep_category": "wgs"}, {"tag": "WGSACUR600", "prep_category": "wgs"}, {"tag": "WGSDPFR400", "prep_category": "wgs"}, {"tag": "WGSDPFR800", "prep_category": "wgs"}, {"tag": "WGSKAPR020", "prep_category": "wgs"}, {"tag": "WGSLIFC030", "prep_category": "wgs"}, {"tag": "WGSLIFC060", "prep_category": "wgs"}, {"tag": "WGSLIFC090", "prep_category": "wgs"}, {"tag": "WGSLIFR400", "prep_category": "wgs"}, {"tag": "WGSLIFR500", "prep_category": "wgs"}, {"tag": "WGSLIFR800", "prep_category": "wgs"}, {"tag": "WGSLIFS120", "prep_category": "wgs"}, {"tag": "WGSLIPC030 (WIP)", "prep_category": "wgs"}, {"tag": "WGSLIPC060 (WIP)", "prep_category": "wgs"}, {"tag": "WGSPCFC015", "prep_category": "wgs"}, {"tag": "WGSPCFC030", "prep_category": "wgs"}, {"tag": "WGSPCFC060", "prep_category": "wgs"}, {"tag": "WGSPCFC090", "prep_category": "wgs"}, {"tag": "WGSPCFR010", "prep_category": "wgs"}, {"tag": "WGSPCFR400", "prep_category": "wgs"}, {"tag": "WGSPCFR500", "prep_category": "wgs"}, {"tag": "WGSPCFR800", "prep_category": "wgs"}, {"tag": "WGSPCFS120", "prep_category": "wgs"}, {"tag": "WGTLIFC030", "prep_category": "wgs"}, {"tag": "WGTPCFC030", "prep_category": "wgs"}, {"tag": "WGXCUSC000", "prep_category": "wgs"}, {"tag": "RNADEPR025", "prep_category": "wts"}, {"tag": "RNADEPR030", "prep_category": "wts"}, {"tag": "RNADEPR050", "prep_category": "wts"}, {"tag": "RNADEPR075", "prep_category": "wts"}, {"tag": "RNAPOAR025", "prep_category": "wts"}, {"tag": "RNAPOAR030", "prep_category": "wts"}, {"tag": "RNAPOAR050", "prep_category": "wts"}, {"tag": "RNAPOAR075", "prep_category": "wts"}, {"tag": "RNAPOAR100", "prep_category": "wts"}, {"tag": "RNLDEPR025", "prep_category": "wts"}, {"tag": "RNLDEPR030", "prep_category": "wts"}, {"tag": "RNLPOAR025", "prep_category": "wts"}, {"tag": "RNLPOAR030", "prep_category": "wts"}, {"tag": "RNLPOAR050", "prep_category": "wts"}, {"tag": "RNLPOAR075", "prep_category": "wts"}, {"tag": "RNLPOAR100", "prep_category": "wts"}, {"tag": "RNXCUSR000", "prep_category": "wts"}]
Traceback (most recent call last):
  File "/home/proj/stage/bin/miniconda3/envs/S_main/bin/cg", line 8, in <module>
    sys.exit(base())
  File "/home/proj/stage/bin/miniconda3/envs/S_main/lib/python3.7/site-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/home/proj/stage/bin/miniconda3/envs/S_main/lib/python3.7/site-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/home/proj/stage/bin/miniconda3/envs/S_main/lib/python3.7/site-packages/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/proj/stage/bin/miniconda3/envs/S_main/lib/python3.7/site-packages/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/proj/stage/bin/miniconda3/envs/S_main/lib/python3.7/site-packages/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/proj/stage/bin/miniconda3/envs/S_main/lib/python3.7/site-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/proj/stage/bin/miniconda3/envs/S_main/lib/python3.7/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/home/proj/stage/bin/miniconda3/envs/S_main/lib/python3.7/site-packages/click/decorators.py", line 27, in new_func
    return f(get_current_context().obj, *args, **kwargs)
  File "/home/proj/stage/bin/miniconda3/envs/S_main/lib/python3.7/site-packages/cg/cli/upload/vogue.py", line 71, in apptags
    upload_vogue_api.load_apptags()
  File "/home/proj/stage/bin/miniconda3/envs/S_main/lib/python3.7/site-packages/cg/meta/upload/vogue.py", line 47, in load_apptags
    self.vogue_api.load_apptags(apptags_for_vogue)
  File "/home/proj/stage/bin/miniconda3/envs/S_main/lib/python3.7/site-packages/cg/apps/vogue.py", line 49, in load_apptags
    self.process.run_command(parameters=load_call)
  File "/home/proj/stage/bin/miniconda3/envs/S_main/lib/python3.7/site-packages/cg/utils/commands.py", line 100, in run_command
    command, check=False, stdout=subprocess.PIPE, stderr=subprocess.PIPE
  File "/home/proj/stage/bin/miniconda3/envs/S_main/lib/python3.7/subprocess.py", line 466, in run
    with Popen(*popenargs, **kwargs) as process:
  File "/home/proj/stage/bin/miniconda3/envs/S_main/lib/python3.7/subprocess.py", line 769, in __init__
    restore_signals, start_new_session)
  File "/home/proj/stage/bin/miniconda3/envs/S_main/lib/python3.7/subprocess.py", line 1516, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: '/home/proj/stage/bin/miniconda3/envs/S_vogue/bin/vogue': '/home/proj/stage/bin/miniconda3/envs/S_vogue/bin/vogue'

ChrOertlin commented on cg/store/api/import_func.py:
```diff
@@ -51,9 +52,7 @@ def import_application_versions(
-        latest_version: models.ApplicationVersion = store.latest_version(
-            application_version.app_tag
-        )
+        latest_version: ApplicationVersion = store.latest_version(application_version.app_tag)
```
I will add these when going over the ApplicationVersion model!

ChrOertlin commented on tests/meta/workflow/conftest.py:
```diff
@@ -199,8 +199,12 @@ def qc_microsalt_context(
-    store.application(tag=MicrosaltAppTags.MWRNXTR003).target_reads = MicrosaltQC.TARGET_READS
-    store.application(tag=MicrosaltAppTags.MWXNXTR003).target_reads = MicrosaltQC.TARGET_READS
+    store.get_application_by_tag(
+        tag=MicrosaltAppTags.MWRNXTR003
```
I checked the constants of CG, currently I can only see three app tags that have a constant. Those are the ones for the MicrosaltAppTags. 

I will start collecting ApplicationTags in a contants file and populate, we can decide on a meeting what and how exactly this should be. 

I can make it part of this PR, then a future single PR to hunt all string application tags throughout cg down and replace them.

diitaz93 commented on tests/store/api/test_findbasicdata.py:
```diff
@@ -123,6 +121,78 @@ def test_get_application_query(base_store: Store):
     assert isinstance(application_query, Query)
 
 
+def test_get_application_by_tag(microbial_store: Store, tag: str = "MWRNXTR003"):
```
Maybe turn this tag value into a fixture

diitaz93 commented on tests/store/api/test_store_import_func.py:
```diff
@@ -82,7 +82,7 @@ def test_versions_are_same(applications_store: Store, application_versions_file:
 
     # GIVEN that the application exists in the database
-    application_obj: models.Application = store.application(version.app_tag)
+    application_obj: models.Application = store.get_application_by_tag(version.app_tag)
```
Do `from models import Application` and replace `models.` throughout this file

diitaz93 commented on tests/store/api/test_store_import_func.py:
```diff
@@ -250,12 +254,16 @@ def test_sync_microbial_orderform_dry_run(microbial_store: Store, microbial_orde
-    inactive_mic_apps_after_when = microbial_store.applications(
-        category=prep_category, archived=True
-    ).count()
+    active_mic_apps_after_when = len(
```
Took me a while to know what it was meant by when, maybe rename to this?
```suggestion
    active_mic_apps_after_sync: int = len(
```

diitaz93 commented on tests/store/api/test_store_import_func.py:
```diff
@@ -288,12 +300,16 @@ def test_sync_microbial_orderform_activate(microbial_store: Store, microbial_ord
-    inactive_mic_apps_after_when = microbial_store.applications(
-        category=prep_category, archived=True
-    ).count()
+    active_mic_apps_after_when = len(
```
same here

diitaz93 commented on tests/store/api/test_store_import_func.py:
```diff
@@ -329,7 +349,11 @@ def test_sync_rml_orderform_inactivate(rml_store: Store, rml_orderform: str, hel
     # THEN the number of active should be less and the number of inactive more than before
-    active_apps_after_when = rml_store.applications(category=prep_category, archived=False).count()
-    inactive_apps_after_when = rml_store.applications(category=prep_category, archived=True).count()
+    active_apps_after_when = len(
```
same here

diitaz93 commented on cg/store/api/findbasicdata.py:
```diff
@@ -21,6 +21,7 @@
 from cg.store.filters.status_bed_filters import apply_bed_filter, BedFilter
```
Maybe add underscores to this file name: `find_basic_data.py`

diitaz93 reviewed (commented):
Very minor comments :)

henrikstranneheim commented on tests/store/filters/test_status_application_filters.py:
```diff
@@ -80,6 +81,28 @@ def test_filter_get_applications_is_archived(
+def test_filter_application_is_not_archived(
+    store_with_an_application_with_and_without_attributes: Store,
+) -> None:
+    """Test to get application by is_archived."""
```
```suggestion
    """Test to get application when not archived."""
```

henrikstranneheim commented on tests/store/filters/test_status_application_filters.py:
```diff
@@ -80,6 +81,28 @@ def test_filter_get_applications_is_archived(
+    """Test to get application by is_archived."""
+    # GIVEN a store with two applications of which one is archived
+
+    # WHEN getting an application by is_archived
```
```suggestion
    # WHEN getting an application that is not archived
```

henrikstranneheim reviewed (approved):
👍

ChrOertlin commented on cg/store/api/findbasicdata.py:
```diff
@@ -32,21 +33,81 @@ class FindBasicDataHandler(BaseHandler):
+
+    def get_applications_by_prep_category(self, prep_category: str) -> List[Application]:
+        """Return applications by prep category."""
+        records: Query = self._get_application_query()
```
Yes, I agree not sure why I did not do this here initially. thanks for the catch!

ChrOertlin commented on tests/meta/orders/test_meta_orders_status.py:
```diff
@@ -533,7 +539,7 @@ def test_store_mip_rna(orders_api, base_store, mip_rna_status_data, ticket: str)
     rna_application = "RNAPOAR025"
```
WIl fix in future PR

ChrOertlin commented:
## Description

### Added

- Tests for Application model functions

### Changed

- Function naming
- Refactored statusDB access to go via filter functions
- Refactored applications() into multiple functions
- Refactored applications() calls into new function calls of smaller functions

### Fixed

- MockApplication to be iterable to conform to new List[Application] return type of get_applications() function

ChrOertlin commented:
stage deploy:
Logging deploy ...
Getting deployer... done.
Getting last commit message and SHA... done.
Getting version of deploy scripts... /home/hiseq.clinical/application_sheet
done.
Log deploy... done.
cg, version 27.0.9


Production deloy:

Logging deploy ...
Getting deployer... done.
Getting last commit message and SHA... done.
Getting version of deploy scripts... /home/hiseq.clinical/application_sheet
done.
Log deploy... done.
cg, version 27.0.9
